org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/copyqualifiedname/XbaseCopyQualifiedNameService.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2013, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.ui.editor.copyqualifiedname;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.common.types.JvmConstructor;
+import org.eclipse.xtext.common.types.JvmExecutable;
+import org.eclipse.xtext.common.types.JvmOperation;
+import org.eclipse.xtext.ui.editor.copyqualifiedname.DefaultCopyQualifiedNameService;
+import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+import org.eclipse.xtext.xbase.XConstructorCall;
+import org.eclipse.xtext.xbase.XExpression;
+import org.eclipse.xtext.xbase.typesystem.IBatchTypeResolver;
+import org.eclipse.xtext.xbase.typesystem.IResolvedTypes;
+import org.eclipse.xtext.xbase.typesystem.override.IResolvedConstructor;
+import org.eclipse.xtext.xbase.typesystem.override.IResolvedExecutable;
+import org.eclipse.xtext.xbase.typesystem.override.IResolvedOperation;
+import org.eclipse.xtext.xbase.typesystem.override.OverrideHelper;
+import org.eclipse.xtext.xbase.typesystem.override.ResolvedFeatures;
+import org.eclipse.xtext.xbase.typesystem.references.LightweightTypeReference;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.FluentIterable;
+import com.google.inject.Inject;
+
+/**
+ * @author Anton Kosyakov - Initial contribution and API
+ * @since 2.4
+ */
+public class XbaseCopyQualifiedNameService extends DefaultCopyQualifiedNameService {
```

Comment:
`org.eclipse.xtext.ui.editor.copyqualifiedname.DefaultCopyQualifiedNameService` is deprecated. We'll have to revisit this code and replace the deprecated API with the correct ones. Corresponding change needs to be made in the `XbaseUiModule#bindCopyQualifiedNameService()` method.

Reply:
Please take a look at the conversion that was applied to the org.eclipse.xtext.naming.DefaultCopyQualifiedNameService - there we use a PolymorphicDispatcher. 
I'm afraid we have to do the same here in order to remain backwards compatible.

Reply:
I'll file a separate jira for this.

Reply:
https://github.com/eclipse/xtext-extras/issues/581